client/src/ingredients/status-banners.js

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+
+const banners = {
+  deprecated: {
+    class: "deprecated-banner",
+    text:
+      "This is an <strong>deprecated</strong> technology. Avoid using it, and update existing code if possible. Be aware that this feature may cease to work at any time."
+  },
+  experimental: {
+    class: "experimental-banner",
+    text:
+      "This is an <strong>experimental</strong> technology. Check the Browser compatibility table carefully before using it in production."
+  },
+  non_standard: {
+    class: "nonstandard-banner",
+    text:
+      "This is an <strong>non-standard</strong> technology. Do not use it on production sites facing the Web: it will not work for every user."
+  }
+};
+
+function StatusBanner({ content }) {
+  return (
+    <div
+      className={`banner ${content.class}`}
+      dangerouslySetInnerHTML={{ __html: content.text }}
+    />
+  );
+}
+
+export function StatusBanners({ status }) {
+  return (
+    <>
+      {status.deprecated && <StatusBanner content={banners["deprecated"]} />}
+      {status.experimental && (
+        <StatusBanner content={banners["experimental"]} />
+      )}
+      {status.non_standard && (
+        <StatusBanner content={banners["non_standard"]} />
+      )}
+    </>
```

Comment:
The point I'm going to make is broken a bit by the fact that maybe I don't understand https://github.com/wbamberg/stumptown-experiment/tree/support-banners fully. So pardon me in advance. 

It seems strange that it's an object and not an array. For two reasons:
* Now the renderer has to own the order of display. I.e. showing the `deprecated` before `non_standard`. 
* Why have stuff that's never carrying any information. 

If the packaged JSON could be this instead:
```json
  "body": [
    {
      "type": "status",
      "value": ["non_standard", "deprecated"],
    },

```
(instead of ...
```json
  "body": [
    {
      "type": "status",
      "value": {
        "non_standard": true,
        "experimental": false,
        "deprecated": true
      }
    },
```

...you get the same information but without the "waste". 

I don't think we're going to get away from having to spell it all out in that `const banners = ` thing you have on line 3 but if it was an array, it could be this instead:

```javascript
const banners = {
  deprecated: "This is an <strong>deprecated</strong> technology. Avoid using it, and update existing code if possible. Be aware that this feature may cease to work at any time.",
  experimental: "This is an <strong>experimental</strong> technology. Check the Browser compatibility table carefully before using it in production.",
  non_standard: "This is an <strong>non-standard</strong> technology. Do not use it on production sites facing the Web: it will not work for every user."
};
export function StatusBanners({ statuses }) {
  return statuses.map(status => {
      const banner = banners[status]
      if (!banner) {
         throw new Error(`Unrecognized status banner '${status}'`);
      }
      return <div key={status}
      className={`banner ${status}-banner`}
      dangerouslySetInnerHTML={{ __html: banner }}
    />
  });
}
```

It's more compact. And it doesn't require mentioning, by name, things like `experimental` in 3 places. And if something strays if you find out about it explicitly instead of it silently being ignored.

Reply:
This suggestion assumes we're OK with the CSS class names having to perfectly match the keys in `scripts/build-json/resolve-status.js` in stumptown-content but I think that's fair.